### DISCUSSION TITLE
how to release an update only for new OS versions without users on older macOS versions getting the "your OS is too old" message?

### DISCUSSION THREAD

**Discussion by core-code** [sparkle-project/Sparkle#2612] [Q&A] [answered]
sorry if this has been covered in the documentation, i could not find anything on this topic.

we had to release a special version of our app exclusively for macOS 15 beta users, to work around another awful Apple bug. 

we properly annotated the special version as requiring macOS 15, so that users without faulty beta software would not be bugged by the update:
```
			<sparkle:minimumSystemVersion>15.0</sparkle:minimumSystemVersion>
```

however now all the users with normal stable OS versions are seeing a dialogue "Your macOS version is too old" when checking for updates. and consequently sending us confused e-mails.

so, how can we release something just for macOS 15 without users on older macOS versions noticing in any way?

**Comment by zorgiepoo** [comment, marked as answer]
> however now all the users with normal stable OS versions are seeing a dialogue "Your macOS version is too old" when checking for updates.

Note this should apply to manual user-initiated update checks, not automatic ones.

> so, how can we release something just for macOS 15 without users on older macOS versions noticing in any way?

I don't think there is a way without setting up channels (easiest), separate appcasts, or appcast item filtering ahead of time. It sounds like this is a temporary restriction for you.
